client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx

```diff
@@ -0,0 +1,183 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import page from 'page';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'calypso/components/data/document-head';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import {
+	emailManagement,
+	emailManagementTitanControlPanelRedirect,
+} from 'calypso/my-sites/email/paths';
+import EmailPlanHeader from 'calypso/my-sites/email/email-management/home/email-plan-header';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import {
+	getEmailPurchaseByDomain,
+	hasEmailSubscription,
+} from 'calypso/my-sites/email/email-management/home/utils';
+import { getDomainsBySite } from 'calypso/state/sites/domains/selectors';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import {
+	hasLoadedSitePurchasesFromServer,
+	isFetchingSitePurchases,
+} from 'calypso/state/purchases/selectors';
+import HeaderCake from 'calypso/components/header-cake';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import QuerySitePurchases from 'calypso/components/data/query-site-purchases';
+import {
+	TITAN_CONTROL_PANEL_CONTEXT_CONFIGURE_CATCH_ALL_EMAIL,
+	TITAN_CONTROL_PANEL_CONTEXT_CONFIGURE_DESKTOP_APP,
+	TITAN_CONTROL_PANEL_CONTEXT_CONFIGURE_INTERNAL_FORWARDING,
+	TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP,
+	TITAN_CONTROL_PANEL_CONTEXT_IMPORT_EMAIL_DATA,
+} from 'calypso/lib/titan/constants';
+import VerticalNav from 'calypso/components/vertical-nav';
+import VerticalNavItemEnhanced from 'calypso/components/vertical-nav/item/enhanced';
+
+class TitanManageMailboxes extends Component {
+	static propTypes = {
+		// Props passed to this component
+		selectedDomainName: PropTypes.string.isRequired,
+
+		// Props injected via connect()
+		currentRoute: PropTypes.string.isRequired,
+		domain: PropTypes.object,
+		hasSubscription: PropTypes.bool.isRequired,
+		isLoadingPurchase: PropTypes.bool.isRequired,
+		purchase: PropTypes.object,
+		selectedSite: PropTypes.object.isRequired,
+
+		// Props injected via localize()
+		translate: PropTypes.func.isRequired,
+	};
+
+	handleBack = () => {
+		const { currentRoute, selectedDomainName, selectedSite } = this.props;
```

Comment:
Should we take a look at `previousRoute` here before we navigate to a hard-coded location?

Reply:
I've never thought of that, and I'm not sure at first glance how that would work. I'll create a card in our Asana board to look into that.